### ISSUE TITLE
[BUG]: tune_model uses n_job=1

### ISSUE THREAD

**Issue by celestinoxp** [pycaret/pycaret#3879]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

After some data engineering, my dataset had perhaps an excess of +/3,000 columns, and +/-2,500 rows. This is obviously reflected in training time, mainly in tuning to search for better models. Using catboost takes 1 minute when using create_model('catboost') and about 2 hours when using tune_model(catboost, n_iter=120)

I went to analyze the log.txt and checked:
2024-01-09 12:00:00,373:INFO:Calculating mean and std
2024-01-09 12:00:00,374:INFO:Creating metrics dataframe
2024-01-09 12:00:00,381:INFO:Finalizing model
2024-01-09 12:00:35,420:INFO:Uploading results into container
2024-01-09 12:00:35,421:INFO:Uploading model into container now
2024-01-09 12:00:35,433:INFO:_master_model_container: 1
2024-01-09 12:00:35,434:INFO:_display_container: 2
2024-01-09 12:00:35,434:INFO:<catboost.core.CatBoostClassifier object at 0x00000279A4569000>
2024-01-09 12:00:35,434:INFO:create_model() successfully completed......................................
2024-01-09 12:00:35,992:INFO:Initializing tune_model()
2024-01-09 12:00:35,992:INFO:tune_model(estimator=<catboost.core.CatBoostClassifier object at 0x00000279A4569000>, fold=None, round=4, n_iter=128, custom_grid=None, optimize=Accuracy, custom_scorer=None, search_library=scikit-learn, search_algorithm=None, early_stopping=False, early_stopping_max_iters=10, choose_better=True, fit_kwargs=None, groups=None, return_tuner=False, verbose=True, tuner_verbose=True, return_train_score=False, kwargs={}, self=<pycaret.classification.oop.ClassificationExperiment object at 0x0000027A355EE200>)
2024-01-09 12:00:35,992:INFO:Checking exceptions
2024-01-09 12:00:36,323:INFO:Copying training dataset
2024-01-09 12:00:36,698:INFO:Checking base model
2024-01-09 12:00:36,699:INFO:Base model : CatBoost Classifier
2024-01-09 12:00:36,703:INFO:Declaring metric variables
2024-01-09 12:00:36,710:INFO:Defining Hyperparameters
2024-01-09 12:00:37,144:INFO:Tuning with n_jobs=1
2024-01-09 12:00:37,144:INFO:Initializing RandomizedSearchCV

My question is: why tuning set n_jobs=1 when i have setup(...n_jobs=-1...use_gpu=True

My cpu have 8 cores/16 threads...

My probem is: gpu not used, all cores not used...  however lightgbm is the unique, uses all cores at 100%

### Reproducible Example

```python
.
```

### Expected Behavior

.

### Actual Results

```python-traceback
.
```

### Installed Versions

<details>
pycaret 3.2.1
</details>
